tests/integration/test_engine_integration.py

```diff
@@ -313,14 +313,19 @@ def test_measurefock_shots(self, setup_eng):
 
     # TODO: when ``shots`` is incorporated into other backends, delete this test
     @pytest.mark.backends("tf", "fock")
-    def test_measurefock_shots_exception(self, setup_eng):
+    def test_measurefock_shots_exception(self, setup_eng, batch_size):
         shots = 5
         eng, p1 = setup_eng(3)
         with p1.context as q:
             ops.MeasureFock() | q
 
         backend_name = eng.backend.__str__()
-        with pytest.raises(NotImplementedError,
-                           match=r"""(Measure|MeasureFock) has not been implemented in {} """
-                                  """for the arguments {{'shots': {}}}""".format(backend_name, shots)):
-            eng.run(p1, run_options={"shots": shots}).samples
+        if batch_size:
+            with pytest.raises(NotImplementedError,
+                               match="Batching cannot be used together with multiple shots."):
+                eng.run(p1, run_options={"shots": shots}).samples
+        else:
+            with pytest.raises(NotImplementedError,
+                               match=r"""(Measure|MeasureFock) has not been implemented in {} """
+                               """for the arguments {{'shots': {}}}""".format(backend_name, shots)):
+                eng.run(p1, run_options={"shots": shots}).samples
```

Comment:
Should this be two separate tests, one for each assert? Seems a bit weird to have a test testing different logic.

Reply:
I guess it could be. The problem was that it takes the environmental variables `BATCHED` and `BATCH_SIZE` for different runs and depending on their values different errors are raised. It's not straight forward to me how to tell a test only to run if BATCHED is True/False. I'll look at it and see if I can figure something out. Two different tests perhaps make more sense. :slightly_smiling_face:

Reply:
I guess I could simply have the `if batch_size:` and `if not batch_size:` in each respective test, and do nothing if `False`, but that would mean running many instances of the tests without actually doing anything, which seems a bit unnecessary.

Reply:
What I normally do is have:
```python
if batch_size:
    pytest.skip("Test only runs for non-batched backends")
```
at the top of the test. Not the best looking, but it usually does the job!